cal/cal/urls.py

```diff
@@ -16,6 +16,9 @@
         'cal.views.complete_with_token', name='complete_with_token'),
     url(r'^accounts/profile', 'cal.views.accounts_profile', name='accounts_profile'),
 
+    url(r'categories/(?P<pk>[0-9]+)/?', 'cal.views.category_detail', name='category_detail'),
+    url(r'tags/(?P<pk>[0-9]+)/?', 'cal.views.tag_detail', name='tag_detail'),
```

Comment:
This is regex that allows you to pass a numerical primary key into the view that is handling the request. You can look at the corresponding view in `views.py` and see that there is an extra `pk` parameter to the function.